lib/src/rules/prefer_const_constructors_in_immutables.dart

```diff
@@ -0,0 +1,146 @@
+    element.library?.name == _META_LIB_NAME;
+
+class PreferConstConstructorsInImmutables extends LintRule {
+  PreferConstConstructorsInImmutables()
+      : super(
+            name: 'prefer_const_constructors_in_immutables',
+            description: desc,
+            details: details,
+            group: Group.style);
+
+  @override
+  AstVisitor getVisitor() => new Visitor(this);
+}
+
+class Visitor extends SimpleAstVisitor {
+  final LintRule rule;
+
+  Visitor(this.rule);
+
+  @override
+  visitConstructorDeclaration(ConstructorDeclaration node) {
+    if (node.body is EmptyFunctionBody &&
+        !node.element.isConst &&
+        !_hasMixin(node.element.enclosingElement) &&
+        _hasImmutableAnnotation(node.element.enclosingElement) &&
+        _hasConstConstructorInvocation(node) &&
+        _hasOnlyConstExpressionsInIntializerList(node)) {
+      rule.reportLintForToken(node.firstTokenAfterCommentAndMetadata);
+    }
+  }
+
+  bool _hasMixin(ClassElement clazz) => clazz.mixins.isNotEmpty;
+
+  bool _hasImmutableAnnotation(ClassElement clazz) {
+    final inheritedAndSelfTypes = _getSelfAndInheritedTypes(clazz.type);
+    final inheritedAndSelfAnnotations = inheritedAndSelfTypes
+        .map((type) => type.element)
+        .expand((c) => c.metadata)
+        .map((m) => m.element);
+    return inheritedAndSelfAnnotations.any(_isImmutable);
+  }
+
+  bool _hasConstConstructorInvocation(ConstructorDeclaration node) {
+    final clazz = node.element.enclosingElement;
+    // construct with super
+    final SuperConstructorInvocation superInvocation = node.initializers
+        .firstWhere((e) => e is SuperConstructorInvocation, orElse: () => null);
+    if (superInvocation != null) return superInvocation.staticElement.isConst;
+    // construct with this
+    final RedirectingConstructorInvocation redirectInvocation = node
+        .initializers
+        .firstWhere((e) => e is RedirectingConstructorInvocation,
+            orElse: () => null);
+    if (redirectInvocation != null)
+      return redirectInvocation.staticElement.isConst;
+    // construct with implicit super()
+    return clazz.supertype.constructors
+        .firstWhere((e) => e.name.isEmpty)
+        .isConst;
+  }
+
+  bool _hasOnlyConstExpressionsInIntializerList(ConstructorDeclaration node) {
+    final typeProvider = node.element.context.typeProvider;
+    final declaredVariables = node.element.context.declaredVariables;
+
+    final listener = new MyAnalysisErrorListener();
+
+    // put a fake const keyword to use ConstantVerifier
+    node.constKeyword = new Token(TokenType.KEYWORD, node.offset);
+    try {
+      final errorReporter = new ErrorReporter(listener, rule.reporter.source);
+      node.accept(new ConstantVerifier(errorReporter, node.element.library,
+          typeProvider, declaredVariables));
+    } finally {
+      // restore const keyword
+      node.constKeyword = null;
+    }
+
+    return !listener.hasConstError;
+  }
```

Comment:
@bwilkerson I have a false positive with this `_hasOnlyConstExpressionsInIntializerList` function and the following code:

```dart
@immutable
class H {
  final f;
  H(f) : f = f ?? f == null; // OK
}
```

If I put `const` on constructor the analyzer triggers the error _In constant expressions, operands of this operator must be of type 'bool', 'num', 'String' or 'null'_ .

Any idea ?